parquet-hadoop/src/main/java/parquet/hadoop/mapred/DeprecatedParquetInputFormat.java

```diff
@@ -39,11 +41,15 @@
   @Override
   public RecordReader<Void, Container<V>> getRecordReader(InputSplit split, JobConf job,
                   Reporter reporter) throws IOException {
-    return new RecordReaderWrapper<V>(realInputFormat, split, job, reporter);
+    return new RecordReaderWrapper<V>(split, job, reporter);
   }
 
   @Override
   public InputSplit[] getSplits(JobConf job, int numSplits) throws IOException {
+    if (isTaskSideMetaData(job)) {
+      return toParquetSplits(super.getSplits(job, numSplits));
+    }
+
```

Comment:
I don't get this? Why the need for the special case?
I would say this is in the wrong place and this wrapper should not have to care about isTaskSideMetaData